internal/k8s/controllers/gateway_controller.go

```diff
@@ -36,8 +36,8 @@ type GatewayReconciler struct {
 //+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=gateways/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=gateways/finalizers,verbs=update
 //+kubebuilder:rbac:groups=core,resources=pods,verbs=list;watch
-//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=list;get;create;watch
-//+kubebuilder:rbac:groups=core,resources=services,verbs=list;get;create;watch
```

Comment:
Good catch, with our current mechanism for e2e testing these sorts of RBAC issues are hard to test as we use the same credentials for the controller as the superuser permissions created by the `kind create` command. Do you mind creating a follow-up issue for tracking purposes of modifying our e2e test suite to use a reduced set of permissions via the generated RBAC policy?